### PR TITLE
CDAP-15821 Whole-file-ingest plugin validation

### PR DESCRIPTION
**Pull request by mlozbin-cybervisiontech**
JIRA: https://issues.cask.co/browse/CDAP-15821
WIKI: https://wiki.cask.co/display/CE/Guidelines+on+Plugin+Validation

In scope of this PR:
- integrated with new validation API

### CODE REVIEW

File: src/main/java/io/cdap/plugin/file/ingest/AbstractFileCopySinkConfig.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        failureCollector.addFailure("Buffer size must be a positive integer.", null)
+          .withConfigProperty(BUFFER_SIZE);
+      }
+    }
+  }
+
+  /*
+   * Additional configurations for the file sink should be implemented in the extended class
+   */
+
```
Comment (albertshau) [inline src/main/java/io/cdap/plugin/file/ingest/AbstractFileCopySinkConfig.java]: nit: remove extra newline

File: src/main/java/io/cdap/plugin/file/ingest/AbstractFileCopySinkConfig.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+      // check if it's null since bufferSize isn't a required field
+      if (bufferSize != null && bufferSize <= 0) {
+        failureCollector.addFailure("Buffer size must be a positive integer.", null)
+          .withConfigProperty(BUFFER_SIZE);
+      }
+    }
+  }
+
+  /*
+   * Additional configurations for the file sink should be implemented in the extended class
```
Comment (albertshau) [inline src/main/java/io/cdap/plugin/file/ingest/AbstractFileCopySinkConfig.java]: this comment doesn't seem accurate

File: src/main/java/io/cdap/plugin/file/ingest/AbstractFileCopySinkConfig.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    super(name);
+    this.basePath = basePath;
+    this.enableOverwrite = enableOverwrite;
+    this.preserveFileOwner = preserveFileOwner;
+    this.bufferSize = bufferSize;
+  }
+
+  public void validate(FailureCollector failureCollector) {
+    try {
+      IdUtils.validateId(referenceName);
```
Comment (albertshau) [inline src/main/java/io/cdap/plugin/file/ingest/AbstractFileCopySinkConfig.java]: the current hydrator-common module has a version that takes a failure collector

**Review by albertshau — COMMENTED** [review]
minor comments, please fix and squash

File: src/main/java/io/cdap/plugin/file/ingest/AbstractFileCopySinkConfig.java (hunk 1 above)
Comment (mlozbin-cybervisiontech) [reply src/main/java/io/cdap/plugin/file/ingest/AbstractFileCopySinkConfig.java]: done

File: src/main/java/io/cdap/plugin/file/ingest/AbstractFileCopySinkConfig.java (hunk 2 above)
Comment (mlozbin-cybervisiontech) [reply src/main/java/io/cdap/plugin/file/ingest/AbstractFileCopySinkConfig.java]: done

File: src/main/java/io/cdap/plugin/file/ingest/AbstractFileCopySinkConfig.java (hunk 3 above)
Comment (mlozbin-cybervisiontech) [reply src/main/java/io/cdap/plugin/file/ingest/AbstractFileCopySinkConfig.java]: done

**Comment by mlozbin-cybervisiontech** [comment]
@albertshau I addressed all your comments and squashed commits, could you, please, merge.

**Pull request merged**